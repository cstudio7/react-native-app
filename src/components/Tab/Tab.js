import React from 'react';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';

export default class Tab extends React.Component {
  render() {
    const { tab } = this.props;
    return (
      <TouchableOpacity
        onPress={tab.onPress}
        style={[styles.tab, tab.isActive ? styles.tabActive : null]}>
        <Text style={styles.tabText}>{tab.name}</Text>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  tab: {
    flex: 1,
    height: 40,
    alignItems: 'center',
    paddingVertical: 11,
    borderBottomWidth: 2,
    borderBottomColor: '#fff'
  },
  tabActive: {
    borderBottomColor: '#000'
  },
  tabText: {
    fontWeight: '600',
    textAlign: 'center'
  }
});
