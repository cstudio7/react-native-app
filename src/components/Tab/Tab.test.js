import React from 'react';
import renderer from 'react-test-renderer';

import Tab from './Tab';

const mockedTab = Object.freeze({
  name: 'Девочки',
  onPress: () => {},
  isActive: true
});

it('renders correctly', () => {
  const tree = renderer.create(<Tab tab={mockedTab} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders active tab', () => {
  const tree = renderer
    .create(<Tab tab={{ ...mockedTab, isActive: false }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
