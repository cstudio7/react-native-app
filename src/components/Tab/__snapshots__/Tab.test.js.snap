// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active tab 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#fff",
      "borderBottomWidth": 2,
      "flex": 1,
      "height": 40,
      "opacity": 1,
      "paddingVertical": 11,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "600",
        "textAlign": "center",
      }
    }
  >
    Женские имена
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomColor": "#000",
      "borderBottomWidth": 2,
      "flex": 1,
      "height": 40,
      "opacity": 1,
      "paddingVertical": 11,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "600",
        "textAlign": "center",
      }
    }
  >
    Женские имена
  </Text>
</View>
`;
